src/button/button.test.ts

```diff
@@ -1,8 +1,88 @@
-import { expect, defineCE, fixture } from '@open-wc/testing';
-import {TapButton} from './'
+import '../../dist/button/index.js';
+import { fixture, expect, html, oneEvent } from '@open-wc/testing';
+import { TapButton } from './index';
+import sinon from 'sinon';
 
 describe('Test tap-button', () => {
-  it('should pass', () => {
-    expect(true).to.equal(true);
+  it('should render button with default properties', async () => {
+    const el = await fixture<TapButton>(html` <tap-button></tap-button>`);
+    expect(el).to.exist;
+    expect(el?.disabled).to.be.false;
   });
+
+  it('should reflects properties to attributes', async () => {
+    const el = await fixture<TapButton>(
+      html` <tap-button disabled type="submit" name="test-name"></tap-button>`,
+    );
+    expect(el?.disabled).to.be.true;
+    expect(el?.getAttribute('type')).to.equal('submit');
+    expect(el?.getAttribute('name')).to.equal('test-name');
+  });
+
+  it('should handles button click', async () => {
+    const el = await fixture<TapButton>(
+      html` <tap-button type="submit"></tap-button>`,
+    );
+    setTimeout(() => el?.click(), 0);
+    const event = await oneEvent(el, 'click');
+    expect(event).to.exist;
+  });
+
+  it('should renders slot content', async () => {
+    const el = await fixture(html`<tap-button>تایید</tap-button>`);
+    const slot = el.shadowRoot!.querySelector('slot');
+    const slotContent = slot!.assignedNodes()[0];
+    expect(slotContent.textContent).to.equal('تایید');
+  });
+
+  it('should be accessible', async () => {
+    const el = await fixture(
+      html`<tap-button label="Test Button"></tap-button>`,
+    );
+
+    await expect(el).to.be.accessible();
+  });
+
+  // todo after add spinner, complete this test
+  // it('should conditionally renders loading state', async () => {
+  // });
```

Comment:
`spinner` is added to button. you can complete this test too.

Reply:
Done